pkg/envoy/sort.go

```diff
@@ -214,17 +214,75 @@ func HeaderMatcherLess(m1, m2 *envoy_api_v2_route.HeaderMatcher) bool {
 		return false
```

Comment:
Since this function is now quite a bit more complex, could you add a comment explaining the general logic? The getters are autogenerated and I'm not sure how to figure out what they fields are actually for.

Reply:
Excellent, exactly what I needed :) Thanks